superset/viz.py

```diff
@@ -460,8 +462,19 @@ def get_df_payload(self, query_obj=None, **kwargs):
                     is_loaded = True
             except Exception as ex:
                 logger.exception(ex)
-                if not self.error_message:
-                    self.error_message = "{}".format(ex)
+
+                error = dataclasses.asdict(
+                    SupersetError(
+                        message="{}".format(ex),
+                        level=ErrorLevel.ERROR,
+                        type=SupersetErrorType.VIZ_GET_DF_ERROR,
+                        extra={},
+                    )
+                )
+                if not self.errors:
```

Comment:
See previous comment which would simplify this to `self.errors.append(error)`.

Reply:
not sure how your previous comment would affect this

Reply:
@etr2460 apologies if I'm misreading the logic but isn't `self.errors` going to be   a list by default and thus rather than:

```python
if not self.errors:
    self.errors = [error]
else:
    self.errors.append(error)
```

this could be:

```python
self.errors.append(error)
```